envs/monkey_zoo/terraform/firewalls.tf

```diff
@@ -96,5 +97,158 @@ resource "google_compute_firewall" "tunneling2-out" {
   }
 
   direction = "EGRESS"
-  destination_ranges = ["10.2.0.0/24"]
+  destination_ranges = ["10.2.0.0/27"]
+}
+
+resource "google_compute_firewall" "allow-tunneling-only-from-islands" {
+  name    = "${local.resource_prefix}allow-tunneling-only-from-islands"
+  network = google_compute_network.monkeyzoo.name
+
+  allow {
+    protocol = "all"
+  }
+
+  direction = "INGRESS"
+  priority = "1000"
+  source_tags = ["island"]
+  target_tags = ["tunneling-9"]
+}
+
+
+resource "google_compute_firewall" "credentials-reuse-in" {
+  name    = "${local.resource_prefix}credentials-reuse-in"
+  network = google_compute_network.credential-reuse.name
+
+  allow {
+    protocol = "all"
+  }
+
+  direction = "INGRESS"
+  priority = "1000"
+  source_ranges = ["0.0.0.0/0"]
```

Comment:
This means that anyone even outside the organization can access it, are you sure?

Reply:
Thats what we already have. Maybe we need to resolve the ones in the production zoo instead

Reply:
This is the "production zoo", terraform defines what it should be